t_custom_branges/test_http_brange.py

```diff
@@ -0,0 +1,528 @@
+            ),
+            param(
+                name="negative",
+                header_value="unknown-value%",
+                expected_status="400",
+            ),
+        ]
+    )
+    def test_http_token_brange(self, name, header_value, expected_status):
+        self._update_tempesta_config(directive="http_token_brange", characters="0x2d-0x7a")
+        self.start_all_services()
+
+        client = self.get_client("deproxy")
+        client.send_request(
+            request=client.create_request(method="GET", headers=[]),
+            expected_status_code=expected_status,
+        )
+        self.assertEqual(
+            client.connection_is_closed(),
+            True if expected_status == "400" else False,
+        )
+
+    @parameterize.expand(
+        [
+            param(
+                name="positive",
+                characters="0x41-0x7a 0x3d",
+                cache_control="unknown=value",
+                expected_status="200",
+            ),
+            param(
+                name="disallow_equal",
+                characters="0x41-0x7a",
+                cache_control="unknown=value",
+                expected_status="400",
+            ),
+            param(
+                name="negative",
+                characters="0x41-0x7a 0x3d",
+                cache_control="unkno%wn=value%",
+                expected_status="400",
+            ),
+        ]
+    )
+    def test_http_qetoken_brange(self, name, characters, cache_control, expected_status):
+        self._update_tempesta_config(directive="http_token_brange", characters=characters)
+        self.start_all_services()
+
+        client = self.get_client("deproxy")
+        client.parsing = False
+        client.send_request(
+            request=client.create_request(method="GET", headers=[("cache-control", cache_control)]),
+            expected_status_code=expected_status,
+        )
+        self.assertEqual(
+            client.connection_is_closed(),
+            True if expected_status == "400" else False,
+        )
+
+    @parameterize.expand(
+        [
+            param(
+                name="positive",
+                characters="0x20-0x7f",
+                date="Mon, 12 Dec 2016 13:59:39 GMT",
+                expected_status="200",
+            ),
+            param(
+                name="negative",
+                characters="0x20 0x2c 0x2e 0x30-0x3b 0x41-0x5a 0x61-0x7a",
+                date="Mon, 12 Dec 2016% 13:59:39 GMT",
+                expected_status="400",
+            ),
+        ]
+    )
+    def test_http_nctl_brange(self, name, characters, date, expected_status):
+        self._update_tempesta_config(directive="http_nctl_brange", characters=characters)
+        self.start_all_services(client=False)
+
+        client = self.get_client("deproxy")
+
+        for header in ["date", "expires", "last-modified", "if-modified-since"]:
+            with self.subTest(msg=f"subTest with {header}' header."):
```

Comment:
This test is will fall because of:

1.  `"expires", "last-modified"` - Response only headers and must be removed from this test.
2.   `date` - Date is not parsed for requests (only for response), but maybe it should, need to discuss.

Reply:
Done. I separated tests for requests/responses.I removed test for `Date` header in requests and updated wiki.